src/components/chip/index.tsx

```diff
@@ -255,22 +255,25 @@ const Chip = ({
               marginRight: Spacings.s1
             };
           }
-          if (rightElement && leftElement) {
-            return {
-              marginHorizontal: 2
-            };
-          }
-          if (iconSource || leftElement) {
-            return {
-              marginLeft: 2,
-              marginRight: Spacings.s3
-            };
-          }
-          if (rightIconSource || rightElement) {
-            return {
-              marginLeft: Spacings.s3,
-              marginRight: 2
-            };
+          if (iconSource || leftElement || rightIconSource || rightElement) {
+            const marginFromElement = Spacings.s1;
+            if ((iconSource || leftElement) && (rightIconSource || rightElement)) {
+              return {
+                marginHorizontal: marginFromElement
+              };
+            }
+            if (iconSource || leftElement) {
+              return {
+                marginLeft: marginFromElement,
+                marginRight: Spacings.s3
+              };
+            }
+            if (rightIconSource || rightElement) {
+              return {
+                marginLeft: Spacings.s3,
+                marginRight: marginFromElement
+              };
+            }
           }
```

Comment:
A small suggestion for making this a little more readable
you can define two consts
```
const addLeftMargin = !!(iconSource || leftElement)
const addRightMargin = !!(rightIconSource || rightElement)
```
Then the conditions will much more readable. Also if there will new elements in either left or right we can simply add them to the const definition and that's it

Last, consider using ternary to make things a little shorter, for example 
```
const marginStyle = {
  marginHorizontal: addLeftMargin && addLeftMargin ? Spacings.1 : 0,
 marginLeft: ...
 marginRight: ... 
}
``

Reply:
Did you mean
`marginHorizontal: addLeftMargin && addRightMargin ? Spacings.s1 : 0`?
About the last part. I don't really understand what you mean, If I use marginHorizontal and margin left or right in the same object marginHorizontal will always be overridden by the left and right margins. So I don't really know how to use them together in the same object.